packages/plugin-ext/src/main/browser/editor/untitled-resource.ts

```diff
@@ -14,39 +14,91 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { ResourceResolver, Resource } from '@theia/core';
+import { injectable, inject } from 'inversify';
+import { TextDocumentContentChangeEvent } from 'vscode-languageserver-protocol';
+import { Resource, ResourceResolver } from '@theia/core';
 import URI from '@theia/core/lib/common/uri';
-import { injectable } from 'inversify';
 import { Schemes } from '../../../common/uri-components';
+import { FileSystem } from '@theia/filesystem/lib/common/filesystem';
+import { FileSystemWatcher } from '@theia/filesystem/lib/browser/filesystem-watcher';
+import { OpenerService } from '@theia/core/lib/browser';
+import { FileResource } from '@theia/filesystem/lib/browser';
 
 const resources = new Map<string, UntitledResource>();
 let index = 0;
+
 @injectable()
 export class UntitledResourceResolver implements ResourceResolver {
-    resolve(uri: URI): Resource | Promise<Resource> {
-        if (uri.scheme === Schemes.UNTITLED) {
-            return resources.get(uri.toString())!;
+
+    @inject(FileSystem)
+    protected readonly fileSystem: FileSystem;
+
+    @inject(FileSystemWatcher)
+    protected readonly fileSystemWatcher: FileSystemWatcher;
+
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+    async resolve(uri: URI): Promise<UntitledResource> {
+        if (uri.scheme !== Schemes.UNTITLED) {
+            throw new Error('The given uri is not untitled file uri: ' + uri);
+        } else {
+            const untitledResource = resources.get(uri.toString());
+            if (!untitledResource) {
+                return createUntitledResource('', '', uri, this.fileSystem, this.fileSystemWatcher);
+            } else {
+                return untitledResource;
+            }
         }
-        throw new Error(`scheme ${uri.scheme} is not '${Schemes.UNTITLED}'`);
     }
 }
 
 export class UntitledResource implements Resource {
+    private fileResource: FileResource;
 
-    constructor(public uri: URI, private content?: string) {
+    constructor(public uri: URI, private content?: string, private fileSystem?: FileSystem, private fileSystemWatcher?: FileSystemWatcher) {
```

Comment:
why not having FileResourceResolver as a member instead of FileSystem & FileSystemWatcher? I see that the creation is the same, shouldn't we delegate to FileResourceResolver?

Reply:
fixed, done